tarofreighter/chain_porter.go

```diff
@@ -462,7 +462,18 @@ func (p *ChainPorter) transferReceiverProof(pkg *sendPackage) error {
 			// addr of the remote party here
 			ctx, cancel := p.WithCtxQuitNoTimeout()
 			defer cancel()
-			err := p.cfg.ProofCourier.DeliverProof(
+
+			// Before attempting to deliver the proof, log that
+			// an attempted delivery is about to occur.
+			err := p.cfg.ExportLog.LogParcelDeliveryAttempt(
```

Comment:
Hmm, I think this should instead live in the implementation of `ProofCourier` we're using right now. Since IIUC, that'll also have the retry logic embedded within it, allowing us to further hide the internal details of retries, etc from the porter.

Reply:
So we'd keep the on disk structure, but then update the current hash mail proof courier to observe this new state.